### PR TITLE
ROIs visualizer

### PR DESCRIPTION
**Pull request by guaje**
This PR adds a feature to visualize **binary** ROI images as contours in horizon. The new parameters needed to use this feature are:

- `--roi_images` (boolean, optional): Enables the representation of binary images as contours.
- `--roi_colors` (rgb color, optional): Color applied to the contours. The default color is red (1, 0, 0).

Here are some examples using this new parameters:

| dipy_horizon --roi_images non-binary.nii.gz binary1.nii.gz binary2.nii.gz | dipy_horizon --roi_colors 0 0 1 --roi_images non-binary.nii.gz binary1.nii.gz binary2.nii.gz |
|---|---|
|Issue1|Issue2|

Additionally, the parameter `--random_colors` has been modified to provide random colors to the ROIs without altering its original functionality. The new values supported by this param are `tracts` and `rois` for the tractograms and the ROIs respectively.

Here are some examples showing the change:

| dipy_horizon --cluster --random_colors --roi_images non-binary.nii.gz binary1.nii.gz binary2.nii.gz *.trk | dipy_horizon --cluster --random_colors tracts --roi_images non-binary.nii.gz binary1.nii.gz binary2.nii.gz *.trk | dipy_horizon --cluster --random_colors rois --roi_images non-binary.nii.gz binary1.nii.gz binary2.nii.gz *.trk |
|---|---|---|
|Issue1|Issue2|Issue2|

### CODE REVIEW

File: dipy/viz/app.py [hunk 1]
```diff
@@ -109,9 +107,14 @@ def __init__(self, tractograms=None, images=None, pams=None,
             small animal data you may need to use something smaller such
             as 2.0. The threshold is in mm. For this parameter to be active
             ``cluster`` should be enabled.
-        random_colors : bool
-            Given multiple tractograms have been included then each tractogram
-            will be shown with different color
+        random_colors : string
```
Comment (skoudoro) [inline dipy/viz/app.py:110]: `str, optional` 

Can you also add the optional missing on the parameter above. Thanks!

File: dipy/viz/app.py [hunk 2]
```diff
@@ -797,9 +881,13 @@ def horizon(tractograms=None, images=None, pams=None,
         small animal data you may need to use something smaller such
         as 2.0. The threshold is in mm. For this parameter to be active
         ``cluster`` should be enabled.
-    random_colors : bool
-        Given multiple tractograms have been included then each tractogram
-        will be shown with different color
+    random_colors : string
```
Comment (skoudoro) [inline dipy/viz/app.py:884]: Doest it means we cannot have both (tracts and roi)?

File: dipy/workflows/viz.py [hunk 3]
```diff
@@ -46,9 +47,14 @@ def run(self, input_files, cluster=False, cluster_thr=15.,
             small animal brains you may need to use something smaller such
             as 2.0. The distance is in mm. For this parameter to be active
             ``cluster`` should be enabled.
-        random_colors : bool, optional
-            Given multiple tractograms have been included then each tractogram
-            will be shown with different color.
+        random_colors : variable str, optional
```
Comment (skoudoro) [inline dipy/workflows/viz.py:50]: Sometimes you request a string and sometimes a list of string.
This is confusing, can you clarify ?

**Review by skoudoro — COMMENTED** [review]
Hi @guaje,

Thank you for this! I still need to play more with it but there is a confusing part (see below). Can you clarify it?
Also, it would be good to add some tests and documentation. 

Thank you !

**Comment by skoudoro** [comment]
Could you also rebate this PR to make sure you have the last change. Thank you

File: dipy/viz/app.py (hunk 1 above)
Comment (guaje) [reply dipy/viz/app.py]: Added

File: dipy/workflows/viz.py (hunk 3 above)
Comment (guaje) [reply dipy/workflows/viz.py]: Updated. It only needs a string.

File: dipy/viz/app.py (hunk 2 above)
Comment (guaje) [reply dipy/viz/app.py:884]: That's the default option. `dipy_horizon --random_colors --roi_images non-binary.nii.gz binary1.nii.gz binary2.nii.gz tract1.trk tract2.trk` will apply a random color to `binary1.nii.gz`, another to `binary2.nii.gz`, another to `tract1.trk` and a different color to `tract2.trk`.

If the random colors are desired only on the tractograms or on the rois, then the respective options should be selected (`tracts` and `rois`).

See usage in the description of the PR. I'm providing an example of the command used to generate each one of the figures.

**Comment by Garyfallidis** [comment]
Thanks @guaje! Merging...

**Comment by Garyfallidis** [comment]
Done!

**Pull request merged**